src/model_repository_manager.cc

```diff
@@ -611,7 +650,17 @@ ModelRepositoryManager::LoadUnloadModel(
   }
 
   bool polled = true;
-  RETURN_IF_ERROR(LoadUnloadModels(models, type, unload_dependents, &polled));
+  while (true) {
+    bool no_parallel_conflict = true;
+    RETURN_IF_ERROR(LoadUnloadModels(
+        models, type, unload_dependents, &polled, &no_parallel_conflict));
+    // Check for parallel load/unload conflict
+    if (no_parallel_conflict) {
+      break;
+    }
+    RevLockGuard<std::mutex> unlock(poll_mu_);
+    std::this_thread::yield();
```

Comment:
To be updated.
It is possible to return a model specific condition variable instead of the model id when a conflict is detected, so the while loop can wait for signal on the condition variable.

Reply:
Update commit: [Refactor parallel load logic](https://github.com/triton-inference-server/core/pull/165/commits/56ce83eb3d33e2253af0084d8ad4efa9bb800ab2)